pkg/operation/care/garden_health.go

```diff
@@ -0,0 +1,254 @@
+		ctx,
+		h.gardenNamespace,
+		requiredVirtualGardenControlPlaneDeployments,
+		requiredVirtualGardenControlPlaneEtcds,
+		condition,
+	); err != nil || exitCondition != nil {
+		return exitCondition, err
+	}
+
+	c := v1beta1helper.UpdatedConditionWithClock(h.clock, condition, gardencorev1beta1.ConditionTrue, "VirtualGardenControlPlaneRunning", "All control plane components are healthy.")
+	return &c, nil
+}
+
+func (h *GardenHealth) checkGardenSystemComponents(ctx context.Context, checker *HealthChecker, condition gardencorev1beta1.Condition) (*gardencorev1beta1.Condition, error) {
+	managedResources := sets.List(requiredGardenRuntimeManagedResources)
+	managedResources = append(managedResources, istio.ManagedResourceNames(true, "virtual-garden-")...)
+
+	if features.DefaultFeatureGate.Enabled(features.HVPA) {
+		managedResources = append(managedResources, hvpa.ManagedResourceName)
+	}
+	if h.isVPAEnabled() {
+		managedResources = append(managedResources, vpa.ManagedResourceControlName)
+	}
+
+	return h.checkManagedResources(ctx, checker, condition, managedResources, "SystemComponentsRunning", "All system components are healthy.")
+}
+
+func (h *GardenHealth) checkVirtualGardenComponents(ctx context.Context, checker *HealthChecker, condition gardencorev1beta1.Condition) (*gardencorev1beta1.Condition, error) {
+	managedResources := sets.List(requiredVirtualGardenManagedResources)
+
+	return h.checkManagedResources(ctx, checker, condition, managedResources, "VirtualGardenComponentsRunning", "All virtual garden components are healthy.")
+}
+
+func (h *GardenHealth) checkManagedResources(
+	ctx context.Context,
+	checker *HealthChecker,
+	condition gardencorev1beta1.Condition,
+	managedResources []string,
+	successReason string,
+	successMessage string,
+) (
+	*gardencorev1beta1.Condition,
+	error,
+) {
+	for _, name := range managedResources {
+		namespace := h.gardenNamespace
+		if sets.New(istio.ManagedResourceNames(true, "virtual-garden-")...).Has(name) {
+			namespace = v1beta1constants.IstioSystemNamespace
+		}
+
+		mr := &resourcesv1alpha1.ManagedResource{}
+		if err := h.runtimeClient.Get(ctx, kubernetesutils.Key(namespace, name), mr); err != nil {
+			if apierrors.IsNotFound(err) {
+				exitCondition := checker.FailedCondition(condition, "ResourceNotFound", err.Error())
+				return &exitCondition, nil
+			}
+			return nil, err
+		}
+
+		if exitCondition := checkManagedResourceForGarden(checker, condition, mr); exitCondition != nil {
+			return exitCondition, nil
+		}
+	}
+	c := v1beta1helper.UpdatedConditionWithClock(h.clock, condition, gardencorev1beta1.ConditionTrue, successReason, successMessage)
+	return &c, nil
+}
+
+func (h *GardenHealth) isVPAEnabled() bool {
+	if h.garden.Spec.RuntimeCluster.Settings != nil &&
+		h.garden.Spec.RuntimeCluster.Settings.VerticalPodAutoscaler != nil &&
+		pointer.BoolDeref(h.garden.Spec.RuntimeCluster.Settings.VerticalPodAutoscaler.Enabled, false) {
+		return true
+	}
+	return false
```

Comment:
What about making it even more concise?

```suggestion
	return h.garden.Spec.RuntimeCluster.Settings != nil &&
		h.garden.Spec.RuntimeCluster.Settings.VerticalPodAutoscaler != nil &&
		pointer.BoolDeref(h.garden.Spec.RuntimeCluster.Settings.VerticalPodAutoscaler.Enabled, false)
```